src/server/common/dbconn.cpp

```diff
@@ -147,31 +184,82 @@ std::unique_ptr<simple_wml::document> dbconn::get_game_history(int player_id, in
 "    select 1 "
 "    from "+db_game_player_info_table_+" player1 "
 "    where game.INSTANCE_UUID = player1.INSTANCE_UUID "
-"      and game.GAME_ID = player1.GAME_ID "
-"      and player1.USER_ID = ? "
-"  ) "
+"      and game.GAME_ID = player1.GAME_ID ";
+
+	if(player_id != 0)
+	{
+		game_history_query += " and player1.USER_ID = ? ";
+		params.emplace_back(player_id);
+	}
+
+	game_history_query += "  ) "
 "  and game.INSTANCE_UUID = player.INSTANCE_UUID "
 "  and game.GAME_ID = player.GAME_ID "
 "  and player.USER_ID != -1 "
-"  and game.END_TIME is not NULL "
-"inner join "+db_game_content_info_table_+" scenario "
+"  and game.END_TIME is not NULL ";
+
+	if(!search_game_name.empty())
+	{
+		game_history_query += " and game.GAME_NAME like concat(?)";
+
+		utils::to_sql_wildcards(search_game_name, false);
```

Comment:
`to_sql_wildcards` escapes `_` and does not support `?`. Is that wanted here as well? Do we tell the user somewhere that they can use wildcards? (We might as well tell them to use SQL wildcards?)

Reply:
`to_sql_wildcards` optionally escapes `_`, and here is set to not do that (the second argument to the method).

Reply:
Alright, what about the rest?

Reply:
I wouldn't expect regular people to know what SQL wildcards are, and underscores in particular are already in a lot of IDs so treating them as a wildcard would be unexpected. Asterisks are at least somewhat more widely known.

Reply:
Do we tell the user somewhere that they can use wildcards?

Reply:
Not currently, no.

Reply:
Updated.

Reply:
Ok, I guess that can be done later. I don't think it makes much of a difference whether we tell users that `*` matches any character and `?` a single one or the same for `%` and `_`. Since those characters match themselves it's not that important if you don't escape them.

I think the conversion makes it more confusing since `%` works anyway and if you write `\*` you get `\%` instead.